Source/HTTP/httpcall.cpp

```diff
@@ -101,6 +101,12 @@ HRESULT CALLBACK HC_CALL::PerfomAsyncProvider(XAsyncOp op, XAsyncProviderData co
         }
         else
         {
+            // If Custom ReponseWriteFunction is specified and compressedResponse is specified reset to default response body write callback
```

Comment:
Won't you need to store the client's custom callback still? So that you can call it when the request completes and you've decompressed the response?

Reply:
Added HCHttpCallResponseBodyWriteFunction temporaryResponseBodyWriteFunction and temporaryResponseBodyWriteFunctionContext fields to the HC_CALL struct to store the client's custom callback if these conditions are met